packages/tailwindcss/src/compat/config/resolve-config.test.ts

```diff
@@ -173,78 +173,87 @@ test('theme keys can reference other theme keys using the theme function regardl
 })
 
 test('theme keys can read from the CSS theme', () => {
-  let theme = new Theme()
-  theme.add('--color-green', 'green')
+  let originalWarn = console.warn
```

Comment:
I'd probably just mock `console` for the entire test file — it'll help cleanup the test and makes adding / testing additional warnings in the future simpler:

```ts
  const warn = vi.spyOn(console, 'log').mockImplementation(() => undefined);

  afterAll(() => warn.mockReset());
```

```ts
    expect(warn).toHaveBeenCalledWith(
       'Using the plugin object parameter as the theme function is deprecated. Please use the `theme` property instead.',
     )
```

Reply:
Not a fan of mocking apis like this too broadly since it can lead to us hiding warnings somewhere where we don't want it. Can put this test into a describe block to scope this and still use the same pattern though 👍 (or can use the `onTestFinished` hook to clean up and have it inside the single test only.. 🤔)